packages/expo-updates/src/Updates.ts

```diff
@@ -115,13 +115,23 @@ export const createdAt: Date | null = ExpoUpdates.commitTime
   ? new Date(ExpoUpdates.commitTime)
   : null;
 
-const isUsingDeveloperTool = !!(manifest as any).developer?.tool;
-const isUsingExpoDevelopmentClient = NativeModulesProxy.ExponentConstants?.appOwnership === 'expo';
-const manualUpdatesInstructions = isUsingExpoDevelopmentClient
-  ? 'To test manual updates, publish your project using `expo publish` and open the published ' +
-    'version in this development client.'
-  : 'To test manual updates, make a release build with `npm run ios --configuration Release` or ' +
-    '`npm run android --variant Release`.';
+/**
+ * In normal development, we block accessing the updates API methods on the JS side, but when developing in
+ * Expo Go or a development client build, the controllers should have control over which API methods should
+ * be allowed.
+ */
+const shouldDeferToNativeForAPIMethodAvailabilityInDevelopment =
+  !!ExpoUpdates.shouldDeferToNativeForAPIMethodAvailabilityInDevelopment;
+
+/**
+ * Developer tool is set when a project is served by `expo start`.
+ */
+const isUsingDeveloperTool =
+  'extra' in manifest ? !!manifest.extra?.expoGo?.developer?.tool : false;
```

Comment:
This should have the same effect, or is there a TypeScript issue?
```suggestion
const isUsingDeveloperTool = !!manifest.extra?.expoGo?.developer?.tool;
```

Reply:
`manifest` is a union type and not all sub-types have the `extra` field, so we must first narrow the union type before accessing `extra`. the other sub-type (the falsey case of this ternary) doesn't have extra/developer/etc.